tools/common/Assembly.cs

```diff
@@ -55,25 +55,31 @@ public partial class Assembly
 
 		public AssemblyDefinition AssemblyDefinition;
 		public Target Target;
-		public bool IsFrameworkAssembly { get { return is_framework_assembly.Value; } }
+		public bool? IsFrameworkAssembly { get { return is_framework_assembly; } }
 		public string FullPath {
 			get {
 				return full_path;
 			}
 			set {
 				full_path = value;
-				if (!is_framework_assembly.HasValue) {
+				if (!is_framework_assembly.HasValue && !string.IsNullOrEmpty (full_path)) {
 					var real_full_path = Target.GetRealPath (full_path);
+#if NET
+					// TODO: Figure out how to determine whether an assembly is a framework assembly or not (but it's not urgent to implement, it's just a performance improvement)
```

Comment:
Tricky, `[AssemblyMetadata]` looked like a good candidate

However we often use `IsFrameworkAssembly` to mean "this was tested and known to work with current platform code and tools". That assumption won't be true anymore since the framework code can be updated independently.

We'll need to audit all call sites in order to determine why the code needs to know if it's a framework (or not) assembly.

Reply:
`IsFrameworkAssembly` wasn't used all that much, since I changed the return value to be nullable I had to update (and audit) every call site in this PR.